backend/routes/projects.py

```diff
@@ -564,31 +565,14 @@ def add_segmentations(project_id, data_id, segmentation_id=None):
         if request_user != data.assigned_user:
             return jsonify(message="Unauthorized access!"), 401
 
-        if request.method == "POST" and segmentation_id == None:
-            segmentation = Segmentation(
-                data_id=data_id, start_time=start_time, end_time=end_time
-            )
-        else:
-            segmentation = Segmentation.query.filter_by(
-                data_id=data_id, id=segmentation_id
-            ).first()
-        segmentation.set_transcription(transcription)
-        values = []
-        for label_name, label_value in annotations.items():
-            app.logger.info(label_name)
-            app.logger.info(label_value)
-            if type(label_value["values"]) is list:
-                for val_id in label_value["values"]:
-                    value = LabelValue.query.filter_by(
-                        id=int(val_id), label_id=label_value["label_id"]
-                    ).first()
-                    values.append(value)
-            else:
-                value = LabelValue.query.filter_by(
-                    id=int(label_value["values"]), label_id=label_value["label_id"]
-                ).first()
-                values.append(value)
-        segmentation.values = values
+        segmentation = generate_segmentation(
```

Comment:
Please check if it is a `PUT` request or `POST` request.

Reply:
I believe we do not have to check for it, the function `generate_segmentation` handles it. If the segmentation_id is None, it implies that the request made is a `POST` request of adding data which the function handles by creating new data. If segmentation_id is **provided**, it implies that the request made is a `PUT` request for updation of the segmentation which the `function` handles by loading the appropriate segment and then adding the changes to it.

Reply:
Agreed. This should work.